doc/NAC/PnacSonicHLD.md

```diff
@@ -0,0 +1,371 @@
+| 21 | 1. Enable NAC feature globally from CLI. 2. Enable NAC on all ports. 3. Initiate PING from PC1 to PC2(or vice versa) in the test topology (say via Etherne16). 4. Switch NAC type from port to mac. 5. Monitor the PING traffic | PING should fail initially, Once NAC type is switched to MAC, then PING should be successful. |
+| 22 | 1. Enable NAC feature globally from CLI. 2. Initiate Supplicant with VALID credentials and allow successful PNAC authentication. 3. Initiate PING from PC1 to PC2(or vice versa) in the test topology. 4. While PING traffic ongoing, modify the supplicant credentials. 5. Re-initiate Supplicant transaction with INVALID credentials. 6. Monitor the PING traffic. | PING should be successful when PNAC is authorized and should fail, when PANC authorization fails. |
+| 23 | Pre-Condition: NAC feature is disabled globally. 1. Initiate PING from PC1 to PC2(or vice versa) - PING should work normally. 2. Enable NAC feature globally. 3. Enable NAC feature on a particular interface(Ethernet16). | Previous PING traffic should be stopped |
+| 24 | Pre-Condition: Let the supplicant be authorized on a particular port of DUT (say Ethernet16). 1. Shutdown Ethernet48 on the Tester, so that Ethernet16 should be Operationally DOWN and Admin State is UP. 2. Start the Ethernet48 on the Tester board | Whenever the port is Operationally DOWN, the corresponding NAC authorization state should be unauthorized. |
+# 8 To be done in future release
+|   Id  |                        Requirement Description                                                                        |
+|------|----------------------------------------------------------------------------------------------------|
+| 1  | The software allows a user to choose, at runtime, between port based authentication and MAC based authentication. |
+| 2 | The software allows a user to assign the VLAN for an authenticated device based on pre-defined configuration |
+| 3 | The software allows a user to place unauthenticated devices in a separate VLAN instead of denying network access (known as MAC bypass) |
+| 4 | The software interfaces with a RADIUS client to get the authentication done from a RADIUS server instead of from a local database |
+| 5 | The software interfaces with a Tacacs+ client to get the authentication done from a Tacacs+ server instead of from a local database |
+| 6 | The software allows a user to specify a list of blacklisted MAC addresses to deny access |
+| 7 | The software allows a user to specify a list of whitelist MAC addresses which will be allowed to access the network without being specifically authenticated.|
+| 8 | The software supports usage of EAP-TLS for secure exchange of credentials across the networks |
+| 9 | The software includes a CLI for a user to monitor the feature. |
+|10 | The software supports SNMP supports for configuration managment. |
```

Comment:
Why do we need SNMP based configuration?

Reply:
We are using IEEE Std 802.1X-2020 for reference. This std does support MIB. To support MIB SNMP is implemented for his feature.

Reply:
I understand that the std supports MIB but we use MIB only for get operations not for configurations. Just wanted to be sure what support we're adding with SNMP.

Reply:
It's for GET operations only.

Reply:
The hostapd implementation uses IEEE 2004 implementation of 802.1X. How are you converging/mapping the 2020 standard with 2004 implementation?

Reply:
As I have mentioned in above comments 802.1.X-2020 std used as reference for SNMP MIB. We are reusing "hostapd" and hence no std. convergence planned for the release.